// semantic-react/lib/collections/breadcrumb/divider.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

var _utilities = require('../../utilities');

var _icon = require('../../elements/icon/icon');

var _icon2 = _interopRequireDefault(_icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toArray(arr) { return Array.isArray(arr) ? arr : Array.from(arr); }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Divider for breadcrumb
 */
var BreadcrumbDivider = (_temp = _class = function (_React$PureComponent) {
    _inherits(BreadcrumbDivider, _React$PureComponent);

    function BreadcrumbDivider() {
        _classCallCheck(this, BreadcrumbDivider);

        return _possibleConstructorReturn(this, (BreadcrumbDivider.__proto__ || Object.getPrototypeOf(BreadcrumbDivider)).apply(this, arguments));
    }

    _createClass(BreadcrumbDivider, [{
        key: 'render',

        /* eslint-enable */

        value: function render() {
            var _props = this.props,
                component = _props.component,
                defaultClasses = _props.defaultClasses,
                children = _props.children,
                other = _objectWithoutProperties(_props, ['component', 'defaultClasses', 'children']);

            other.className = (0, _classnames2.default)(other.className, this.getClasses());
            var Component = component;

            // If divider has icon child return only icon
            if ((0, _utilities.hasFirstChild)(children, BreadcrumbDivider.Components.Icon)) {
                var _React$Children$toArr = _react2.default.Children.toArray(children),
                    _React$Children$toArr2 = _toArray(_React$Children$toArr),
                    iconChild = _React$Children$toArr2[0],
                    otherChilds = _React$Children$toArr2.slice(1);

                return _react2.default.cloneElement(iconChild, {
                    className: (0, _classnames2.default)(iconChild.props.className, this.getClasses())
                });
            } else {
                return _react2.default.createElement(
                    Component,
                    other,
                    children
                );
            }
        }

        /* eslint-disable */

    }, {
        key: 'getClasses',
        value: function getClasses() {
            return {
                divider: this.props.defaultClasses
            };
        }
    }]);

    return BreadcrumbDivider;
}(_react2.default.PureComponent), _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps), _class.Components = {
    Icon: _icon2.default
}, _temp);
exports.default = BreadcrumbDivider;
