// semantic-react/lib/modules/dropdown/dropdownmenu.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _reactEventListener = require('react-event-listener');

var _reactEventListener2 = _interopRequireDefault(_reactEventListener);

var _elementType = require('react-prop-types/lib/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

var _utilities = require('../../utilities');

var _animation = require('../../animation/animation');

var _animation2 = _interopRequireDefault(_animation);

var _animationProps = require('../../animation/animationProps');

var _animationProps2 = _interopRequireDefault(_animationProps);

var _dropdownelement = require('./dropdownelement');

var _dropdownelement2 = _interopRequireDefault(_dropdownelement);

var _icon = require('./../../elements/icon/icon');

var _icon2 = _interopRequireDefault(_icon);

var _text = require('./../../elements/simple/text');

var _text2 = _interopRequireDefault(_text);

var _menu = require('./../../views/menu/menu');

var _menu2 = _interopRequireDefault(_menu);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Dropdown menu with animations
 */
var DropdownMenu = (_temp2 = _class = function (_React$PureComponent) {
    _inherits(DropdownMenu, _React$PureComponent);

    function DropdownMenu() {
        var _ref;

        var _temp, _this, _ret;

        _classCallCheck(this, DropdownMenu);

        for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = arguments[_key];
        }

        return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = DropdownMenu.__proto__ || Object.getPrototypeOf(DropdownMenu)).call.apply(_ref, [this].concat(args))), _this), _this.onOutsideDropdownClick = function (e) {
            var _this$props = _this.props,
                active = _this$props.active,
                onRequestClose = _this$props.onRequestClose;

            if (!active) {
                return;
            }
            var dropdownElement = _reactDom2.default.findDOMNode(_this);
            if (dropdownElement) {
                if (!(0, _utilities.isNodeInRoot)(e.target, dropdownElement)) {
                    onRequestClose();
                }
            }
        }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    /* eslint-disable */

    /* eslint-enable */

    /**
     * Outside dropdown click
     * @param {React.MouseEvent} e
     */


    _createClass(DropdownMenu, [{
        key: 'renderMenuIcon',


        /**
         * Renders dropdown icon
         * @returns {*}
         */
        value: function renderMenuIcon() {
            if (typeof this.props.component === 'function') {
                return null;
            } else {
                return _react2.default.createElement(DropdownMenu.Components.Icon, { name: this.props.icon });
            }
        }

        /**
         * Renders dropdown text
         * @returns {*}
         */

    }, {
        key: 'renderMenuText',
        value: function renderMenuText() {
            if (typeof this.props.component === 'function' || !this.props.label) {
                return null;
            } else {
                return _react2.default.createElement(
                    DropdownMenu.Components.Text,
                    null,
                    this.props.label
                );
            }
        }

        /**
         * Render menu
         */

    }, {
        key: 'renderMenu',
        value: function renderMenu() {
            /* eslint-disable no-use-before-define */
            var _props = this.props,
                active = _props.active,
                children = _props.children,
                menuComponent = _props.menuComponent,
                menuValue = _props.menuValue,
                onMenuChange = _props.onMenuChange,
                onMenuItemClick = _props.onMenuItemClick;
            /* eslint-enable no-use-before-define */

            var MenuComponent = menuComponent || DropdownMenu.Components.Menu;

            return _react2.default.createElement(
                MenuComponent,
                { key: 'menu',
                    menuValue: menuValue,
                    onMenuChange: onMenuChange,
                    onMenuItemClick: onMenuItemClick
                },
                children
            );
        }

        /**
         * Render
         * @returns {JSX.Element}
         */

    }, {
        key: 'render',
        value: function render() {
            /* eslint-disable no-use-before-define */
            var _props2 = this.props,
                active = _props2.active,
                enter = _props2.enter,
                leave = _props2.leave,
                enterDuration = _props2.enterDuration,
                leaveDuration = _props2.leaveDuration,
                children = _props2.children,
                icon = _props2.icon,
                label = _props2.label,
                menuComponent = _props2.menuComponent,
                menuValue = _props2.menuValue,
                onMenuChange = _props2.onMenuChange,
                onMenuItemClick = _props2.onMenuItemClick,
                onRequestClose = _props2.onRequestClose,
                onAnimationStyle = _props2.onAnimationStyle,
                other = _objectWithoutProperties(_props2, ['active', 'enter', 'leave', 'enterDuration', 'leaveDuration', 'children', 'icon', 'label', 'menuComponent', 'menuValue', 'onMenuChange', 'onMenuItemClick', 'onRequestClose', 'onAnimationStyle']);
            /* eslint-enable no-use-before-define */

            return _react2.default.createElement(
                DropdownMenu.Components.DropdownElement,
                _extends({}, other, {
                    active: active
                }),
                _react2.default.createElement(_reactEventListener2.default, {
                    target: document,
                    onMouseDownCapture: this.onOutsideDropdownClick,
                    onTouchStartCapture: this.onOutsideDropdownClick }),
                this.renderMenuText(),
                this.renderMenuIcon(),
                _react2.default.createElement(
                    _animation2.default,
                    {
                        enter: enter,
                        leave: leave,
                        enterDuration: enterDuration,
                        leaveDuration: leaveDuration
                    },
                    active && this.renderMenu()
                )
            );
        }
    }]);

    return DropdownMenu;
}(_react2.default.PureComponent), _class.defaultProps = _extends({}, _dropdownelement2.default.defaultProps, _animationProps2.default.defaultProps, {
    enter: 'slide down in',
    leave: 'slide down out',
    enterDuration: 200,
    leaveDuration: 200,
    active: false,
    icon: 'dropdown',
    onMenuItemClick: function onMenuItemClick() {},
    onMenuChange: function onMenuChange() {},
    onRequestClose: function onRequestClose() {}
}), _class.Components = {
    Icon: _icon2.default,
    Text: _text2.default,
    Menu: _menu2.default,
    DropdownElement: _dropdownelement2.default
}, _temp2);
exports.default = DropdownMenu;
