// semantic-react/lib/modules/dropdown/select.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _reactEventListener = require('react-event-listener');

var _reactEventListener2 = _interopRequireDefault(_reactEventListener);

var _dropdownelement = require('./dropdownelement');

var _dropdownelement2 = _interopRequireDefault(_dropdownelement);

var _label = require('./../../elements/label/label');

var _label2 = _interopRequireDefault(_label);

var _icon = require('./../../elements/icon/icon');

var _icon2 = _interopRequireDefault(_icon);

var _header = require('./../../elements/header/header');

var _header2 = _interopRequireDefault(_header);

var _menu = require('./../../views/menu/menu');

var _menu2 = _interopRequireDefault(_menu);

var _utilities = require('../../utilities');

var _animation = require('../../animation/animation');

var _animation2 = _interopRequireDefault(_animation);

var _animationProps = require('../../animation/animationProps');

var _animationProps2 = _interopRequireDefault(_animationProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Select is the dropdown where options could be selected, either single or multiple.
 * Also supports search
 */
var Select = (_temp = _class = function (_React$PureComponent) {
    _inherits(Select, _React$PureComponent);

    /* eslint-enable */

    function Select(props) {
        _classCallCheck(this, Select);

        /**
         * Menu reference
         */
        var _this = _possibleConstructorReturn(this, (Select.__proto__ || Object.getPrototypeOf(Select)).call(this, props));

        _this.onOutsideDropdownClick = function (event) {
            var _this$props = _this.props,
                active = _this$props.active,
                onRequestClose = _this$props.onRequestClose;

            if (!active) {
                return;
            }
            var element = _reactDom2.default.findDOMNode(_this);
            if (element) {
                if (!(0, _utilities.isNodeInRoot)(event.target, element)) {
                    onRequestClose();
                }
            }
        };

        _this.onMenuItemClick = function (value) {
            var _this$props2 = _this.props,
                search = _this$props2.search,
                selected = _this$props2.selected,
                multiple = _this$props2.multiple,
                onSelectChange = _this$props2.onSelectChange,
                onSearchStringChange = _this$props2.onSearchStringChange;

            var optionsCount = _this.menuRef ? _react2.default.Children.count(_this.menuRef.props.children) : 0;
            if (selected.indexOf(value) === -1) {
                if (multiple) {
                    // Append value for multiple
                    onSelectChange([].concat(_toConsumableArray(selected), [value]), optionsCount);
                } else {
                    // replace for single
                    onSelectChange([value], optionsCount);
                }

                // Intelligently handle multiple select here:
                // Do not request close if selecting and there are more than 1 element left in menu
                // Set focus to search box if searchable
                // Close if menu has only 1 element
                if (multiple && _this.menuRef) {
                    if (optionsCount <= 1) {
                        // Clean search string
                        onSearchStringChange('');
                    } else if (search && _this.searchRef) {
                        // we have few more elements here, put focus if searchable
                        _this.searchRef.focus();
                    }
                } else if (_this.searchRef) {
                    // always clean search string for single selection dropdowns
                    onSearchStringChange('');
                }
            }
        };

        _this.onSearchInputChange = function (event) {
            var onSearchStringChange = _this.props.onSearchStringChange;


            onSearchStringChange(event.target.value);
        };

        _this.onSearchInputKeyDown = function (event) {
            var _this$props3 = _this.props,
                allowAdditions = _this$props3.allowAdditions,
                searchString = _this$props3.searchString,
                onSelectChange = _this$props3.onSelectChange,
                selected = _this$props3.selected,
                multiple = _this$props3.multiple,
                onSearchStringChange = _this$props3.onSearchStringChange;

            switch (event.which) {
                // Enter
                case 13:
                    // only do something if we have search results available and not displaying not results message
                    if (_this.menuRef) {
                        if (!_this.noResultsMessageRef && _react2.default.Children.count(_this.menuRef.props.children) > 0) {
                            // get the first children
                            var child = _react2.default.Children.toArray(_this.menuRef.props.children)[0];
                            if (child && child.props.value) {
                                // enter should do the same as menu item click
                                _this.onMenuItemClick(child.props.value);
                            }
                        }
                        if (allowAdditions) {
                            if (multiple) {
                                // Append value for multiple
                                var values = [].concat(_toConsumableArray(selected), [searchString]);
                                onSelectChange(values, values.length);
                                onSearchStringChange('');
                            } else {
                                // replace for single
                                onSelectChange([searchString], 1);
                                onSearchStringChange('');
                            }
                        }
                    }
                    break;
                // Backspace
                case 8:
                    if (searchString === '') {
                        if (multiple && selected.length > 0) {
                            onSelectChange([].concat(_toConsumableArray(selected.slice(0, -1))));
                        }
                    }
                    break;
            }
        };

        _this.menuRef = null;
        _this.searchRef = null;
        _this.noResultsMessageRef = null;

        _this.labelStyle = { display: 'block' };
        return _this;
    }

    /* eslint-disable */


    _createClass(Select, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            var _props = this.props,
                active = _props.active,
                search = _props.search;

            if (search && this.searchRef && active) {
                this.searchRef.focus();
            }
        }
    }, {
        key: 'componentDidUpdate',
        value: function componentDidUpdate() {
            var _props2 = this.props,
                active = _props2.active,
                search = _props2.search;

            if (search && this.searchRef && active) {
                this.searchRef.focus();
            }
        }

        /**
         * Handler for outside click
         */


        /**
         * Handler for menu item click
         * @param value
         */

    }, {
        key: 'onLabelCloseIconClick',


        /**
         * Handler for close label click
         * @param value
         */
        value: function onLabelCloseIconClick(value) {
            var _props3 = this.props,
                selected = _props3.selected,
                onSelectChange = _props3.onSelectChange;

            var optionsCount = this.menuRef ? _react2.default.Children.count(this.menuRef.props.children) : 0;
            var index = selected.indexOf(value);
            if (index !== -1) {
                onSelectChange([].concat(_toConsumableArray(selected.filter(function (val) {
                    return val !== value;
                }))), optionsCount);
            }
        }

        /**
         * Handler for search input change
         * @param {React.KeyboardEvent} event
         */


        /**
         * Handler for search input key events
         * @param {React.KeyboardEvent} event
         */

    }, {
        key: 'renderDropdownInput',


        /**
         * Renders dropdown hidden input
         */
        value: function renderDropdownInput() {
            var _props4 = this.props,
                name = _props4.name,
                selected = _props4.selected;

            var value = selected.join(',');

            return _react2.default.createElement('input', { name: name,
                type: 'hidden',
                value: value });
        }

        /**
         * Renders dropdown labels for multiple type dropdowns
         */

    }, {
        key: 'renderDropdownLabels',
        value: function renderDropdownLabels() {
            var _this2 = this;

            var _props5 = this.props,
                multiple = _props5.multiple,
                selected = _props5.selected;
            // Selection labels should appear only for multiple dropdowns

            if (!multiple) {
                return null;
            }

            return _react2.default.Children.map(this.props.children, function (child) {
                // Process only option or option like childs and if it's selected
                if (child && child.props && selected.indexOf(child.props.value) !== -1) {
                    return _react2.default.createElement(
                        _animation2.default,
                        {
                            enter: 'in scale',
                            leave: false,
                            enterDuration: 200,
                            runOnMount: true
                        },
                        _react2.default.createElement(
                            Select.Components.Label,
                            { component: 'a',
                                key: 'label-' + child.props.value,
                                style: _this2.labelStyle
                            },
                            child.props.children,
                            _react2.default.createElement(Select.Components.Icon, { name: 'close',
                                onClick: _this2.onLabelCloseIconClick.bind(_this2, child.props.value) })
                        )
                    );
                }
            });
        }

        /**
         * Render dropdown placeholder text
         */

    }, {
        key: 'renderDropdownText',
        value: function renderDropdownText() {
            /* eslint-disable no-use-before-define */
            var _props6 = this.props,
                multiple = _props6.multiple,
                placeholder = _props6.placeholder,
                search = _props6.search,
                selected = _props6.selected,
                searchString = _props6.searchString,
                selection = _props6.selection;
            /* eslint-enable no-use-before-define */

            // Render placeholder if not selected anything or multiple

            if (searchString.length === 0 && (selected.length === 0 || multiple)) {
                if (typeof placeholder !== 'undefined') {
                    // Selection type should use default text, non selection text
                    return _react2.default.createElement(
                        'div',
                        { className: selection ? 'default text' : 'text' },
                        placeholder
                    );
                } else {
                    return null;
                }
            } else {
                // Single selection here
                // Render selected children in text div here if selected anything
                var content = _react2.default.createElement('div', { className: 'text filtered' });
                // Do not render when performing search
                if (!search || !searchString) {
                    // traverse in childs, find necessary node
                    _react2.default.Children.forEach(this.props.children, function (child) {
                        if (child && child.props && typeof child.props.value !== 'undefined' && selected.indexOf(child.props.value) !== -1) {
                            content = _react2.default.createElement(
                                'div',
                                { className: 'text' },
                                child.props.children
                            );
                        }
                    });
                }
                return content;
            }
        }

        /**
         * Renders dropdown icon
         */

    }, {
        key: 'renderDropdownIcon',
        value: function renderDropdownIcon() {
            var icon = this.props.icon;

            return _react2.default.createElement(Select.Components.Icon, { name: icon });
        }

        /**
         * Renders search input
         */

    }, {
        key: 'renderSearchInput',
        value: function renderSearchInput() {
            var _this3 = this;

            var _props7 = this.props,
                search = _props7.search,
                searchGlyphWidth = _props7.searchGlyphWidth,
                searchString = _props7.searchString,
                searchPosition = _props7.searchPosition,
                multiple = _props7.multiple;

            // Do not render if not searchable

            if (!search) {
                return null;
            }

            var searchWidth = searchGlyphWidth * searchString.length;
            // single selection dropdown shouldn't apply width style
            var style = searchWidth && searchPosition === 'dropdown' && multiple ? { width: searchWidth + 'em' } : {};
            if (searchPosition === 'dropdown') {
                return _react2.default.createElement('input', { autoComplete: 'off',
                    className: 'search',
                    key: 'searchInput',
                    onChange: this.onSearchInputChange,
                    onKeyDown: this.onSearchInputKeyDown,
                    ref: function ref(_ref) {
                        return _this3.searchRef = _ref;
                    },
                    style: style,
                    tabIndex: 0,
                    value: searchString });
            } else {
                // Search in menu has slightly different layout
                return _react2.default.createElement(
                    'div',
                    { className: 'ui icon search input' },
                    _react2.default.createElement(Select.Components.Icon, { name: 'search' }),
                    _react2.default.createElement('input', { key: 'searchInput',
                        onChange: this.onSearchInputChange,
                        onKeyDown: this.onSearchInputKeyDown,
                        placeholder: 'Search...',
                        ref: function ref(_ref2) {
                            return _this3.searchRef = _ref2;
                        },
                        tabIndex: 0,
                        type: 'text',
                        value: searchString })
                );
            }
        }

        /**
         * Renders search header if specified
         */

    }, {
        key: 'renderSearchHeader',
        value: function renderSearchHeader() {
            var _props8 = this.props,
                search = _props8.search,
                searchHeader = _props8.searchHeader;

            if (search && searchHeader) {
                return _react2.default.createElement(
                    Select.Components.Header,
                    { key: 'searchHeader' },
                    searchHeader
                );
            } else {
                return null;
            }
        }

        /**
         * Filters children options to exclude selected elements or elements which doesn't match to search
         */

    }, {
        key: 'renderFilteredChildren',
        value: function renderFilteredChildren() {
            var _props9 = this.props,
                search = _props9.search,
                searchIgnoreCase = _props9.searchIgnoreCase,
                selected = _props9.selected,
                searchString = _props9.searchString,
                multiple = _props9.multiple;

            var searchRegex = new RegExp(searchString, searchIgnoreCase ? 'gi' : 'g');

            var newChildren = [];
            _react2.default.Children.forEach(this.props.children, function (child) {
                // only process option like childs
                if (!child) {
                    return;
                }
                if (typeof child.props.value !== 'undefined') {
                    var match = true;
                    if (search && searchString !== '') {
                        // value could be int or string. In case of int convert it to string
                        var value = typeof child.props.value === 'number' ? Number.parseInt(child.props.value) : child.props.value;
                        match = searchRegex.test(value) || searchRegex.test(child.props.children);
                    }

                    // Match for non search selection will be always true
                    if (match) {
                        // For multiple type render only non selected options
                        if (multiple && selected.indexOf(child.props.value) === -1) {
                            newChildren.push(_react2.default.cloneElement(child, {
                                key: child.key ? child.key : child.props.value,
                                // need to pass menu value, otherwise menu will not bind handler
                                menuValue: child.props.value
                            }));
                        } else if (!multiple) {
                            // for single selection render all options but add selected value
                            newChildren.push(_react2.default.cloneElement(child, {
                                active: selected.indexOf(child.props.value) !== -1,
                                key: child.key ? child.key : child.props.value,
                                // need to pass menu value, otherwise menu will not bind handler
                                menuValue: child.props.value
                            }));
                        }
                    }
                } else if (searchString === '') {
                    // need to pass non option like childs, but only if search string is empty
                    newChildren.push(_react2.default.cloneElement(child, {
                        key: child.key ? child.key : child.props.value
                    }));
                }
            });
            return newChildren;
        }

        /**
         * Render menu
         */

    }, {
        key: 'renderMenu',
        value: function renderMenu() {
            var _this4 = this;

            /* eslint-disable no-use-before-define */
            var _props10 = this.props,
                active = _props10.active,
                search = _props10.search,
                searchPosition = _props10.searchPosition,
                searchHeader = _props10.searchHeader,
                searchString = _props10.searchString,
                searchNoResultsMessage = _props10.searchNoResultsMessage,
                allowAdditions = _props10.allowAdditions,
                allowAdditionsMessage = _props10.allowAdditionsMessage,
                other = _objectWithoutProperties(_props10, ['active', 'search', 'searchPosition', 'searchHeader', 'searchString', 'searchNoResultsMessage', 'allowAdditions', 'allowAdditionsMessage']);
            /* eslint-enable no-use-before-define */
            // make new array for menu childrens


            var menuChildren = [];
            // render search and header in menu
            if (search && searchPosition === 'menu') {
                if (searchHeader) {
                    menuChildren.push(this.renderSearchHeader());
                }
                menuChildren.push(this.renderSearchInput());
            }

            var filteredChild = this.renderFilteredChildren();
            // Display no results message instead of children if needed
            if ((!filteredChild || filteredChild.length === 0) && search && searchString != '') {
                filteredChild = [_react2.default.createElement(
                    'div',
                    { className: 'message',
                        key: 'noResultsMessage',
                        ref: function ref(_ref3) {
                            return _this4.noResultsMessageRef = _ref3;
                        }
                    },
                    allowAdditions ? allowAdditionsMessage : searchNoResultsMessage
                )]; // eslint-disable-line
            }

            menuChildren = menuChildren.concat(filteredChild);

            return _react2.default.createElement(
                Select.Components.Menu,
                {
                    key: 'menu',
                    onMenuItemClick: this.onMenuItemClick,
                    ref: function ref(_ref4) {
                        return _this4.menuRef = _ref4;
                    }
                },
                menuChildren
            );
        }
    }, {
        key: 'render',
        value: function render() {
            /* eslint-disable no-use-before-define */
            var _props11 = this.props,
                active = _props11.active,
                children = _props11.children,
                enter = _props11.enter,
                leave = _props11.leave,
                enterDuration = _props11.enterDuration,
                leaveDuration = _props11.leaveDuration,
                icon = _props11.icon,
                name = _props11.name,
                search = _props11.search,
                searchPosition = _props11.searchPosition,
                searchHeader = _props11.searchHeader,
                searchString = _props11.searchString,
                searchGlyphWidth = _props11.searchGlyphWidth,
                searchIgnoreCase = _props11.searchIgnoreCase,
                searchNoResultsMessage = _props11.searchNoResultsMessage,
                allowAdditionsMessage = _props11.allowAdditionsMessage,
                allowAdditions = _props11.allowAdditions,
                placeholder = _props11.placeholder,
                selected = _props11.selected,
                selection = _props11.selection,
                multiple = _props11.multiple,
                onSelectChange = _props11.onSelectChange,
                onRequestClose = _props11.onRequestClose,
                onSearchStringChange = _props11.onSearchStringChange,
                other = _objectWithoutProperties(_props11, ['active', 'children', 'enter', 'leave', 'enterDuration', 'leaveDuration', 'icon', 'name', 'search', 'searchPosition', 'searchHeader', 'searchString', 'searchGlyphWidth', 'searchIgnoreCase', 'searchNoResultsMessage', 'allowAdditionsMessage', 'allowAdditions', 'placeholder', 'selected', 'selection', 'multiple', 'onSelectChange', 'onRequestClose', 'onSearchStringChange']);
            /* eslint-enable no-use-before-define */

            other.className = (0, _classnames2.default)(other.className, this.getClasses());

            return _react2.default.createElement(
                Select.Components.DropdownElement,
                _extends({}, other, {
                    active: active
                }),
                _react2.default.createElement(_reactEventListener2.default, { target: document,
                    onMouseDown: this.onOutsideDropdownClick,
                    onTouchStart: this.onOutsideDropdownClick }),
                this.renderDropdownInput(),
                this.renderDropdownLabels(),
                this.renderDropdownText(),
                this.renderDropdownIcon(),
                search && searchPosition === 'dropdown' && this.renderSearchInput(),
                _react2.default.createElement(
                    _animation2.default,
                    {
                        enter: enter,
                        leave: leave,
                        enterDuration: enterDuration,
                        leaveDuration: leaveDuration
                    },
                    active && this.renderMenu()
                )
            );
        }
    }, {
        key: 'getClasses',
        value: function getClasses() {
            return {
                search: this.props.search,
                selection: this.props.selection,
                multiple: this.props.multiple
            };
        }
    }]);

    return Select;
}(_react2.default.PureComponent), _class.defaultProps = _extends({}, _dropdownelement2.default.defaultProps, _animationProps2.default.defaultProps, {
    active: false,
    icon: 'dropdown',
    search: false,
    searchGlyphWidth: 1.0714,
    searchPosition: 'dropdown',
    searchIgnoreCase: true,
    searchNoResultsMessage: 'No Results found.',
    allowAdditionsMessage: 'Press enter to add.',
    searchString: '',
    selection: true,
    selected: [],
    multiple: false,
    allowAdditions: false,
    enter: "slide down in",
    leave: "slide down out",
    enterDuration: 200,
    leaveDuration: 200,
    onSelectChange: function onSelectChange() {},
    onRequestClose: function onRequestClose() {},
    onSearchStringChange: function onSearchStringChange() {}
}), _class.Components = {
    Label: _label2.default,
    Icon: _icon2.default,
    Header: _header2.default,
    Menu: _menu2.default,
    DropdownElement: _dropdownelement2.default
}, _temp);
exports.default = Select;
