// semantic-react/lib/modules/progress/progress.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _utilities = require('../../utilities');

var _bar = require('./bar');

var _bar2 = _interopRequireDefault(_bar);

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var validProps = {
    attached: ['top', 'bottom']
};

var Progress = (_temp = _class = function (_React$PureComponent) {
    _inherits(Progress, _React$PureComponent);

    function Progress() {
        _classCallCheck(this, Progress);

        return _possibleConstructorReturn(this, (Progress.__proto__ || Object.getPrototypeOf(Progress)).apply(this, arguments));
    }

    _createClass(Progress, [{
        key: 'getChildContext',

        /* eslint-enable */

        value: function getChildContext() {
            return {
                isProgressChild: true
            };
        }

        /* eslint-disable */

    }, {
        key: 'renderBar',
        value: function renderBar() {
            /* eslint-disable no-use-before-define */
            var _props = this.props,
                active = _props.active,
                attached = _props.attached,
                children = _props.children,
                color = _props.color,
                component = _props.component,
                defaultClasses = _props.defaultClasses,
                disabled = _props.disabled,
                error = _props.error,
                indicating = _props.indicating,
                inverted = _props.inverted,
                size = _props.size,
                success = _props.success,
                warning = _props.warning,
                other = _objectWithoutProperties(_props, ['active', 'attached', 'children', 'color', 'component', 'defaultClasses', 'disabled', 'error', 'indicating', 'inverted', 'size', 'success', 'warning']);
            /* eslint-enable no-use-before-define */


            other.progress = this.getProgress(other.progress);

            return _react2.default.createElement(Progress.Components.Bar, _extends({
                key: 'progressBar'
            }, other));
        }
    }, {
        key: 'render',
        value: function render() {
            /* eslint-disable no-use-before-define */
            var _props2 = this.props,
                active = _props2.active,
                attached = _props2.attached,
                children = _props2.children,
                color = _props2.color,
                component = _props2.component,
                defaultClasses = _props2.defaultClasses,
                disabled = _props2.disabled,
                duration = _props2.duration,
                error = _props2.error,
                indicating = _props2.indicating,
                inverted = _props2.inverted,
                size = _props2.size,
                success = _props2.success,
                warning = _props2.warning,
                progress = _props2.progress,
                other = _objectWithoutProperties(_props2, ['active', 'attached', 'children', 'color', 'component', 'defaultClasses', 'disabled', 'duration', 'error', 'indicating', 'inverted', 'size', 'success', 'warning', 'progress']);
            /* eslint-enable no-use-before-define */

            // add classnames


            other.className = (0, _classnames2.default)(this.props.className, this.getClasses());

            return _react2.default.createElement(this.props.component, other, [this.renderBar(), this.props.children]);
        }
    }, {
        key: 'getClasses',
        value: function getClasses() {
            var classes = {
                // default
                ui: this.props.defaultClasses,

                // position based props
                attached: this.props.attached,

                // types

                // states
                active: this.props.active,
                disabled: this.props.disabled,
                error: this.props.error,
                success: this.props.success,
                warning: this.props.warning,

                // variations
                indicating: this.props.indicating,
                inverted: this.props.inverted,

                // component
                progress: this.props.defaultClasses
            };

            // string types
            classes[this.props.color] = !!this.props.color;
            classes[this.props.size] = !!this.props.size;

            return (0, _utilities.validateClassProps)(classes, this.props, validProps);
        }
    }, {
        key: 'getProgress',
        value: function getProgress(progress) {
            if (progress > 100) {
                return 100;
            } else if (progress < 0) {
                return 0;
            } else {
                return progress;
            }
        }
    }]);

    return Progress;
}(_react2.default.PureComponent), _class.childContextTypes = {
    isProgressChild: _propTypes2.default.bool
}, _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps, {
    duration: 300
}), _class.Components = {
    Bar: _bar2.default
}, _temp);
exports.default = Progress;
