// semantic-react/lib/animation/animation.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _TransitionGroup = require('react-transition-group/TransitionGroup');

var _TransitionGroup2 = _interopRequireDefault(_TransitionGroup);

var _animationChildren = require('./animationChildren');

var _animationChildren2 = _interopRequireDefault(_animationChildren);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function FirstChild(props) {
    var childrenArray = _react2.default.Children.toArray(props.children);
    return childrenArray[0] || null;
}

/**
 * Animation component using semantic-ui CSS transitions
 */
var SemanticCSSTransition = (_temp2 = _class = function (_React$Component) {
    _inherits(SemanticCSSTransition, _React$Component);

    function SemanticCSSTransition() {
        var _ref;

        var _temp, _this, _ret;

        _classCallCheck(this, SemanticCSSTransition);

        for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = arguments[_key];
        }

        return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = SemanticCSSTransition.__proto__ || Object.getPrototypeOf(SemanticCSSTransition)).call.apply(_ref, [this].concat(args))), _this), _this.wrapChildren = function (child) {
            var _this$props = _this.props,
                enterDuration = _this$props.enterDuration,
                leaveDuration = _this$props.leaveDuration,
                enter = _this$props.enter,
                leave = _this$props.leave,
                onEnter = _this$props.onEnter,
                onLeave = _this$props.onLeave,
                runOnMount = _this$props.runOnMount;

            return _react2.default.createElement(
                _animationChildren2.default,
                {
                    enterDuration: enterDuration,
                    leaveDuration: leaveDuration,
                    enter: enter,
                    leave: leave,
                    onEnter: onEnter,
                    onLeave: onLeave,
                    runOnMount: runOnMount
                },
                child
            );
        }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(SemanticCSSTransition, [{
        key: 'render',
        value: function render() {
            var _props = this.props,
                enterDuration = _props.enterDuration,
                leaveDuration = _props.leaveDuration,
                enter = _props.enter,
                leave = _props.leave,
                onEnter = _props.onEnter,
                onLeave = _props.onLeave,
                runOnMount = _props.runOnMount,
                other = _objectWithoutProperties(_props, ['enterDuration', 'leaveDuration', 'enter', 'leave', 'onEnter', 'onLeave', 'runOnMount']);

            return _react2.default.createElement(_TransitionGroup2.default, _extends({}, other, {
                component: FirstChild,
                childFactory: this.wrapChildren }));
        }
    }]);

    return SemanticCSSTransition;
}(_react2.default.Component), _class.defaultProps = _extends({}, _animationChildren2.default.defaultProps), _temp2);
exports.default = SemanticCSSTransition;
