// semantic-react/lib/views/comment/comment.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _image = require('./../../elements/image/image');

var _image2 = _interopRequireDefault(_image);

var _content = require('./../../elements/simple/content');

var _content2 = _interopRequireDefault(_content);

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Comment = (_temp = _class = function (_React$PureComponent) {
    _inherits(Comment, _React$PureComponent);

    function Comment() {
        _classCallCheck(this, Comment);

        return _possibleConstructorReturn(this, (Comment.__proto__ || Object.getPrototypeOf(Comment)).apply(this, arguments));
    }

    _createClass(Comment, [{
        key: 'renderChildren',

        /* eslint-enable */

        value: function renderChildren() {
            var commentChildren = [];
            var contentChildren = [];

            _react2.default.Children.forEach(this.children, function (child) {
                // remove the Image default classes
                if (child.type === Comment.Components.Image) {
                    // eslint-disable-line
                    commentChildren.push(_react2.default.cloneElement(child, {
                        component: 'a',
                        avatar: true,
                        key: 'commentAvatar',
                        defaultClasses: false
                    }, child.children));
                } else {
                    return contentChildren.push(child);
                }
            });

            commentChildren.push(_react2.default.createElement(
                Comment.Components.Content,
                { key: 'commentContent' },
                contentChildren
            ));

            return commentChildren;
        }

        /* eslint-disable */

    }, {
        key: 'render',
        value: function render() {
            var _props = this.props,
                children = _props.children,
                className = _props.className,
                component = _props.component,
                defaultClasses = _props.defaultClasses,
                other = _objectWithoutProperties(_props, ['children', 'className', 'component', 'defaultClasses']);

            other.className = (0, _classnames2.default)(className, { comment: true });

            return _react2.default.createElement(component, other, this.renderChildren());
        }
    }]);

    return Comment;
}(_react2.default.PureComponent), _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps), _class.Components = {
    Image: _image2.default,
    Content: _content2.default
}, _temp);
exports.default = Comment;
