// semantic-react/lib/views/feed/event.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _image = require('./../../elements/image/image');

var _image2 = _interopRequireDefault(_image);

var _images = require('./../../elements/image/images');

var _images2 = _interopRequireDefault(_images);

var _icon = require('./../../elements/icon/icon');

var _icon2 = _interopRequireDefault(_icon);

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Event = (_temp = _class = function (_React$PureComponent) {
    _inherits(Event, _React$PureComponent);

    function Event() {
        _classCallCheck(this, Event);

        return _possibleConstructorReturn(this, (Event.__proto__ || Object.getPrototypeOf(Event)).apply(this, arguments));
    }

    _createClass(Event, [{
        key: 'renderLabel',

        /* eslint-enable */

        value: function renderLabel() {
            var labelChildren = [];

            _react2.default.Children.forEach(this.children, function (child, index) {
                // remove the Image default classes
                if ((child.type === Event.Components.Image || child.type === Event.Components.Icon) && index === 0) {
                    // eslint-disable-line
                    if (child.type === Event.Components.Image) {
                        // eslint-disable-line
                        labelChildren.push(_react2.default.cloneElement(child, {
                            key: 'eventLabel',
                            image: false,
                            defaultClasses: false
                        }, child.children));
                    } else {
                        labelChildren.push(child);
                    }
                }
            });

            return _react2.default.createElement(
                'div',
                { className: 'label' },
                labelChildren
            );
        }

        /* eslint-disable */

    }, {
        key: 'renderContent',
        value: function renderContent() {
            var contentChildren = [];

            _react2.default.Children.forEach(this.children, function (child, index) {
                if (child.type === Event.Components.Images) {
                    // eslint-disable-line
                    contentChildren.push(_react2.default.cloneElement(child, {
                        defaultClasses: false,
                        className: 'extra images'
                    }));
                } else if (child.type !== Event.Components.Image && child.type !== Event.Components.Icon) {
                    // eslint-disable-line
                    contentChildren.push(child);
                } else if ((child.type === Event.Components.Image || child.type === Event.Components.Icon) && index > 0) {
                    // eslint-disable-line
                    contentChildren.push(child);
                }
            });

            return _react2.default.createElement(
                'div',
                { className: 'content' },
                contentChildren
            );
        }
    }, {
        key: 'render',
        value: function render() {
            var _props = this.props,
                children = _props.children,
                className = _props.className,
                component = _props.component,
                defaultClasses = _props.defaultClasses,
                other = _objectWithoutProperties(_props, ['children', 'className', 'component', 'defaultClasses']);

            other.className = (0, _classnames2.default)(className, { event: true });

            return _react2.default.createElement(component, other, [this.renderLabel(children), this.renderContent(children)]);
        }
    }]);

    return Event;
}(_react2.default.PureComponent), _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps), _class.Components = {
    Image: _image2.default,
    Images: _images2.default,
    Icon: _icon2.default
}, _temp);
exports.default = Event;
