// semantic-react/lib/collections/form/field.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _utilities = require('../../utilities');

var _checkbox = require('./../../modules/checkbox/checkbox');

var _checkbox2 = _interopRequireDefault(_checkbox);

var _checkboxfields = require('./../../modules/checkbox/checkboxfields');

var _checkboxfields2 = _interopRequireDefault(_checkboxfields);

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Field = (_temp = _class = function (_React$PureComponent) {
    _inherits(Field, _React$PureComponent);

    function Field() {
        _classCallCheck(this, Field);

        return _possibleConstructorReturn(this, (Field.__proto__ || Object.getPrototypeOf(Field)).apply(this, arguments));
    }

    _createClass(Field, [{
        key: 'renderLabel',

        /* eslint-enable */

        value: function renderLabel(label) {
            return _react2.default.createElement(
                'label',
                { key: label + 'Label' },
                label
            );
        }

        /* eslint-disable */

    }, {
        key: 'render',
        value: function render() {
            /* eslint-disable no-use-before-define */
            var _props = this.props,
                children = _props.children,
                className = _props.className,
                component = _props.component,
                defaultClasses = _props.defaultClasses,
                grouped = _props.grouped,
                inline = _props.inline,
                label = _props.label,
                required = _props.required,
                state = _props.state,
                width = _props.width,
                other = _objectWithoutProperties(_props, ['children', 'className', 'component', 'defaultClasses', 'grouped', 'inline', 'label', 'required', 'state', 'width']);
            /* eslint-enable no-use-before-define */


            other.className = (0, _classnames2.default)(className, this.getClasses());

            return _react2.default.createElement(component, other, [(0, _utilities.hasFirstChild)(children, Field.Components.Checkbox) || (0, _utilities.hasFirstChild)(children, Field.Components.CheckboxFields) || !label || label == '' ? null : this.renderLabel(this.props.label), children]);
        }
    }, {
        key: 'getClasses',
        value: function getClasses() {
            var classes = {
                disabled: this.props.state === 'disabled',
                error: this.props.state === 'error',

                required: this.props.required,
                inline: this.props.inline,
                grouped: this.props.grouped
            };

            if (this.props.width) {
                if (this.props.width > 0 && this.props.width <= 16) {
                    classes[_utilities.Numbers[this.props.width] + ' wide'] = true;
                }
            }

            classes.field = this.props.defaultClasses;

            return classes;
        }
    }]);

    return Field;
}(_react2.default.PureComponent), _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps), _class.Components = {
    Checkbox: _checkbox2.default,
    CheckboxFields: _checkboxfields2.default
}, _temp);
exports.default = Field;
