// semantic-react/lib/modules/search/search.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _icon = require('./../../elements/icon/icon');

var _icon2 = _interopRequireDefault(_icon);

var _results = require('./results');

var _results2 = _interopRequireDefault(_results);

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

var _animation = require('../../animation/animation');

var _animation2 = _interopRequireDefault(_animation);

var _animationProps = require('../../animation/animationProps');

var _animationProps2 = _interopRequireDefault(_animationProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/*
3 support results types:

{
    title: 'test',
    description: 'testing'
}

{
    animal: ['cat', 'dog']
}

['test', 'test']
*/
var Search = (_temp = _class = function (_React$PureComponent) {
    _inherits(Search, _React$PureComponent);

    /* eslint-enable */

    function Search(props) {
        _classCallCheck(this, Search);

        var _this = _possibleConstructorReturn(this, (Search.__proto__ || Object.getPrototypeOf(Search)).call(this, props));

        _this.onBlur = function () {
            if (_this.state.focus) {
                _this.setState({
                    focus: false
                });
            }
        };

        _this.onChange = function (e) {
            _this.props.onChange(e);
        };

        _this.onFocus = function () {
            _this.setState({
                focus: true
            });
        };

        _this.onSearchClick = function (e, child) {
            _this.props.onSearchClick(e, child);
        };

        _this.state = {
            focus: false
        };
        return _this;
    }

    /* eslint-disable */


    _createClass(Search, [{
        key: 'renderInput',
        value: function renderInput() {
            return _react2.default.createElement(
                'div',
                { className: this.getClassesInput(),
                    key: 'searchInput'
                },
                _react2.default.createElement('input', { className: 'prompt',
                    onBlur: this.onBlur,
                    onChange: this.onChange,
                    onFocus: this.onFocus,
                    placeholder: this.props.placeholder,
                    type: 'text',
                    value: this.props.value }),
                this.renderInputIcon()
            );
        }
    }, {
        key: 'renderInputIcon',
        value: function renderInputIcon() {
            if (!this.props.icon) return null;

            return _react2.default.createElement(Search.Components.Icon, { name: this.props.icon });
        }
    }, {
        key: 'renderResults',
        value: function renderResults() {
            var resultProps = {
                key: 'searchResults',
                emptyHeader: this.props.emptyHeader,
                emptyMessage: this.props.emptyMessage,
                onSearchClick: this.onSearchClick,
                results: this.state.focus ? this.props.results : [],
                search: this.props.value || ''
            };

            var _props = this.props,
                enter = _props.enter,
                leave = _props.leave,
                enterDuration = _props.enterDuration,
                leaveDuration = _props.leaveDuration;


            return _react2.default.createElement(
                _animation2.default,
                {
                    enter: enter,
                    leave: leave,
                    enterDuration: enterDuration,
                    leaveDuration: leaveDuration
                },
                this.state.focus && this.props.value !== '' && _react2.default.createElement(Search.Components.Results, _extends({}, resultProps, { key: 'results' }))
            );
        }
    }, {
        key: 'render',
        value: function render() {
            /* eslint-disable no-use-before-define */
            var _props2 = this.props,
                children = _props2.children,
                className = _props2.className,
                component = _props2.component,
                defaultClasses = _props2.defaultClasses,
                emptyHeader = _props2.emptyHeader,
                emptyMessage = _props2.emptyMessage,
                initialAnimation = _props2.initialAnimation,
                enterAnimation = _props2.enterAnimation,
                icon = _props2.icon,
                leaveAnimation = _props2.leaveAnimation,
                loading = _props2.loading,
                onChange = _props2.onChange,
                onSearchClick = _props2.onSearchClick,
                placeholder = _props2.placeholder,
                results = _props2.results,
                onAnimationStyle = _props2.onAnimationStyle,
                value = _props2.value,
                other = _objectWithoutProperties(_props2, ['children', 'className', 'component', 'defaultClasses', 'emptyHeader', 'emptyMessage', 'initialAnimation', 'enterAnimation', 'icon', 'leaveAnimation', 'loading', 'onChange', 'onSearchClick', 'placeholder', 'results', 'onAnimationStyle', 'value']);
            /* eslint-enable no-use-before-define */

            other.className = (0, _classnames2.default)(this.props.className, this.getClasses());
            var Component = component;
            return _react2.default.createElement(
                Component,
                other,
                this.renderInput(),
                ',',
                this.renderResults()
            );
        }
    }, {
        key: 'getClassesInput',
        value: function getClassesInput() {
            return (0, _classnames2.default)('ui input', {
                icon: Boolean(this.props.icon)
            });
        }
    }, {
        key: 'getClasses',
        value: function getClasses() {
            return {
                ui: this.props.defaultClasses,
                search: this.props.defaultClasses,

                loading: this.props.loading,
                focus: this.state.focus,

                category: !Array.isArray(this.props.results)
            };
        }
    }]);

    return Search;
}(_react2.default.PureComponent), _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps, _animationProps2.default.defaultProps, {
    enter: 'scale in',
    leave: 'scale out',
    enterDuration: 200,
    leaveDuration: 200,
    icon: 'search',
    onSearchClick: function noop() {},
    placeholder: 'Search...',
    value: ''
}), _class.Components = {
    Results: _results2.default,
    Icon: _icon2.default
}, _temp);
exports.default = Search;
