// semantic-react/lib/elements/list/list.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _utilities = require('../../utilities');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var validProps = {
    aligned: ['top', 'middle', 'bottom'],
    floated: ['right', 'left']
};

var List = (_temp = _class = function (_React$PureComponent) {
    _inherits(List, _React$PureComponent);

    function List() {
        _classCallCheck(this, List);

        return _possibleConstructorReturn(this, (List.__proto__ || Object.getPrototypeOf(List)).apply(this, arguments));
    }

    _createClass(List, [{
        key: 'getChildContext',
        value: function getChildContext() {
            return {
                isListChild: true,
                isLinkList: this.props.link
            };
        }
    }, {
        key: 'render',
        value: function render() {
            /* eslint-disable no-use-before-define */
            var _props = this.props,
                component = _props.component,
                defaultClasses = _props.defaultClasses,
                aligned = _props.aligned,
                animated = _props.animated,
                celled = _props.celled,
                floated = _props.floated,
                horizontal = _props.horizontal,
                inverted = _props.inverted,
                link = _props.link,
                relaxed = _props.relaxed,
                selection = _props.selection,
                size = _props.size,
                type = _props.type,
                other = _objectWithoutProperties(_props, ['component', 'defaultClasses', 'aligned', 'animated', 'celled', 'floated', 'horizontal', 'inverted', 'link', 'relaxed', 'selection', 'size', 'type']);
            /* eslint-enable no-use-before-define */

            other.className = (0, _classnames2.default)(this.props.className, this.getClasses());
            var Component = component;
            return _react2.default.createElement(Component, other);
        }
    }, {
        key: 'getClasses',
        value: function getClasses() {
            var classes = {
                // default
                ui: this.props.defaultClasses && !this.context.isListChild,

                // types
                bulleted: this.props.type === 'bulleted',
                horizontal: this.props.horizontal,
                link: this.props.link,
                ordered: this.props.type === 'ordered',

                // variations
                animated: this.props.animated,
                celled: this.props.celled && this.props.celled !== 'divided',
                divided: this.props.celled === 'divided',
                inverted: this.props.inverted,
                relaxed: this.props.relaxed,
                selection: this.props.selection,
                aligned: this.props.aligned,

                // component
                list: this.props.defaultClasses
            };

            classes[this.props.size] = !!this.props.size;

            return (0, _utilities.validateClassProps)(classes, this.props, validProps);
        }
    }]);

    return List;
}(_react2.default.PureComponent), _class.contextTypes = {
    isListChild: _propTypes2.default.bool
}, _class.childContextTypes = {
    isListChild: _propTypes2.default.bool,
    isLinkList: _propTypes2.default.bool
}, _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps), _temp);
exports.default = List;
