// semantic-react/lib/elements/step/step.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _utilities = require('../../utilities');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _elementType = require('react-prop-types/lib/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

var _icon = require('../../elements/icon/icon');

var _icon2 = _interopRequireDefault(_icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var validProps = {};

var Step = (_temp = _class = function (_React$PureComponent) {
    _inherits(Step, _React$PureComponent);

    function Step() {
        _classCallCheck(this, Step);

        return _possibleConstructorReturn(this, (Step.__proto__ || Object.getPrototypeOf(Step)).apply(this, arguments));
    }

    _createClass(Step, [{
        key: 'render',

        /* eslint-enable */

        value: function render() {
            /* eslint-disable no-use-before-define */
            var _props = this.props,
                active = _props.active,
                children = _props.children,
                component = _props.component,
                completed = _props.completed,
                defaultClasses = _props.defaultClasses,
                description = _props.description,
                disabled = _props.disabled,
                icon = _props.icon,
                iconComponent = _props.iconComponent,
                link = _props.link,
                title = _props.title,
                other = _objectWithoutProperties(_props, ['active', 'children', 'component', 'completed', 'defaultClasses', 'description', 'disabled', 'icon', 'iconComponent', 'link', 'title']);
            /* eslint-enable no-use-before-define */

            var Component = component;
            if (Component === 'div' && this.props.link) {
                Component = 'a';
                other.href = this.props.link;
            }

            var IconComponent = iconComponent || Step.Components.Icon;
            other.className = (0, _classnames2.default)(other.className, this.getClasses());

            return _react2.default.createElement(
                Component,
                other,
                icon && _react2.default.createElement(IconComponent, { name: icon }),
                _react2.default.createElement(
                    'div',
                    { className: 'content' },
                    title && _react2.default.createElement(
                        'div',
                        { className: 'title' },
                        title
                    ),
                    description && _react2.default.createElement(
                        'div',
                        { className: 'description' },
                        description
                    )
                ),
                children
            );
        }

        /* eslint-disable */

    }, {
        key: 'getClasses',
        value: function getClasses() {
            var classes = {
                step: this.props.defaultClasses,

                active: this.props.active,
                completed: this.props.completed,
                disabled: this.props.disabled
            };

            return (0, _utilities.validateClassProps)(classes, this.props, validProps);
        }
    }]);

    return Step;
}(_react2.default.PureComponent), _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps), _class.Components = {
    Icon: _icon2.default
}, _temp);
exports.default = Step;
