// semantic-react/lib/modules/dimmer/dimmer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _animation = require('../../animation/animation');

var _animation2 = _interopRequireDefault(_animation);

var _animationProps = require('../../animation/animationProps');

var _animationProps2 = _interopRequireDefault(_animationProps);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _utilities = require('../../utilities');

var _defaultProps = require('../../defaultProps');

var _defaultProps2 = _interopRequireDefault(_defaultProps);

var _content = require('./../../elements/simple/content');

var _content2 = _interopRequireDefault(_content);

var _loader = require('./../../elements/loader/loader');

var _loader2 = _interopRequireDefault(_loader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Dimmer
 */
var Dimmer = (_temp = _class = function (_React$PureComponent) {
    _inherits(Dimmer, _React$PureComponent);

    function Dimmer() {
        _classCallCheck(this, Dimmer);

        return _possibleConstructorReturn(this, (Dimmer.__proto__ || Object.getPrototypeOf(Dimmer)).apply(this, arguments));
    }

    _createClass(Dimmer, [{
        key: 'getChildContext',

        /* eslint-enable */

        value: function getChildContext() {
            return {
                isDimmerChild: true
            };
        }

        /* eslint-disable */


        // This forces <Content> to add a centered div below it.

    }, {
        key: 'componentDidMount',
        value: function componentDidMount() {
            var _props = this.props,
                active = _props.active,
                page = _props.page;

            if (active && page) {
                document.body.classList.add('dimmable', 'dimmed');
            }
        }
    }, {
        key: 'componentDidUpdate',
        value: function componentDidUpdate() {
            var _props2 = this.props,
                active = _props2.active,
                page = _props2.page;

            if (active && page) {
                document.body.classList.add('dimmable', 'dimmed');
            }
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            var page = this.props.page;

            if (page) {
                document.body.classList.remove('dimmable', 'dimmed');
            }
        }

        /**
         * Render dimmer children
         */

    }, {
        key: 'renderChildren',
        value: function renderChildren() {
            var _props3 = this.props,
                children = _props3.children,
                noWrapChildren = _props3.noWrapChildren;

            if (noWrapChildren) {
                return children;
            } else {
                return (0, _utilities.hasChild)(children, Dimmer.Components.Loader) ? children : _react2.default.createElement(
                    Dimmer.Components.Content,
                    null,
                    children
                );
            }
        }
    }, {
        key: 'render',
        value: function render() {
            // Note: Need to consume closePortal since dimmer could be called from portal-ed components (modal/popup)
            /* eslint-disable no-use-before-define */
            var _props4 = this.props,
                active = _props4.active,
                children = _props4.children,
                component = _props4.component,
                defaultClasses = _props4.defaultClasses,
                enterDuration = _props4.enterDuration,
                leaveDuration = _props4.leaveDuration,
                enter = _props4.enter,
                leave = _props4.leave,
                page = _props4.page,
                inverted = _props4.inverted,
                noWrapChildren = _props4.noWrapChildren,
                closePortal = _props4.closePortal,
                other = _objectWithoutProperties(_props4, ['active', 'children', 'component', 'defaultClasses', 'enterDuration', 'leaveDuration', 'enter', 'leave', 'page', 'inverted', 'noWrapChildren', 'closePortal']);
            /* eslint-enable no-use-before-define */

            var Component = component;
            other.className = (0, _classnames2.default)(other.className, this.getClasses());

            return _react2.default.createElement(
                _animation2.default,
                {
                    enter: enter,
                    leave: leave,
                    enterDuration: enterDuration,
                    leaveDuration: leaveDuration,
                    runOnMount: true
                },
                active && _react2.default.createElement(
                    Component,
                    _extends({}, other, {
                        key: 'dimmer'
                    }),
                    this.renderChildren()
                )
            );
        }
    }, {
        key: 'getClasses',
        value: function getClasses() {
            return {
                ui: this.props.defaultClasses,
                active: this.props.active,
                page: this.props.page,
                inverted: this.props.inverted,
                dimmer: this.props.defaultClasses
            };
        }
    }]);

    return Dimmer;
}(_react2.default.PureComponent), _class.childContextTypes = {
    isDimmerChild: _propTypes2.default.bool
}, _class.defaultProps = _extends({}, _defaultProps2.default.defaultProps, _animationProps2.default.defaultProps, {
    enterDuration: 500,
    leaveDuration: 300,
    enter: 'fade in',
    leave: 'fade out',
    active: false,
    page: false,
    inverted: false,
    noWrapChildren: false
}), _class.Components = {
    Content: _content2.default,
    Loader: _loader2.default
}, _temp);
exports.default = Dimmer;
