// semantic-react/lib/elements/header/subheader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _header = require('./header');

var _header2 = _interopRequireDefault(_header);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Sub header, same as header but with 'sub' class name
 * Deprecated
 */
var SubHeader = (_temp = _class = function (_React$PureComponent) {
    _inherits(SubHeader, _React$PureComponent);

    function SubHeader() {
        _classCallCheck(this, SubHeader);

        return _possibleConstructorReturn(this, (SubHeader.__proto__ || Object.getPrototypeOf(SubHeader)).apply(this, arguments));
    }

    _createClass(SubHeader, [{
        key: 'render',

        /* eslint-enable */

        value: function render() {
            var _props = this.props,
                component = _props.component,
                children = _props.children,
                defaultClasses = _props.defaultClasses,
                other = _objectWithoutProperties(_props, ['component', 'children', 'defaultClasses']);

            var Component = component || SubHeader.Components.Header;
            other.className = (0, _classnames2.default)(other.className, { sub: defaultClasses });
            return _react2.default.createElement(
                Component,
                other,
                children
            );
        }

        /* eslint-disable */

    }]);

    return SubHeader;
}(_react2.default.PureComponent), _class.defaultProps = {
    defaultClasses: true,
    item: false
}, _class.Components = {
    Header: _header2.default
}, _temp);
exports.default = SubHeader;
